creme/sketch/bricks.py

```diff
@@ -0,0 +1,165 @@
+#    Copyright (C) 2022  Hybird
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+################################################################################
+
+from random import randint
+
+from django.utils.translation import gettext as _
+
+from creme.creme_core.gui.bricks import Brick
+
+
+class ChartBrick(Brick):
+    """
+    Base class for bricks containing D3js charts.
+    """
+    template_name = "sketch/bricks/chart.html"
+
+    def get_chart_props(self, context):
+        """
+        Properties to set colors, fonts, sizes or animations of a chart
+        """
+        return {}
+
+    def get_chart_data(self, context):
+        """
+        Raw data for the chart. Must be implemented !
+        """
+        raise NotImplementedError()
+
+    def detailview_display(self, context):
+        return self._render(
+            self.get_template_context(
+                context,
+                data=self.get_chart_data(context),
+                props=self.get_chart_props(context)
+            )
+        )
+
+    def home_display(self, context):
+        return self.detailview_display(context)
+
+
+class BarChartBrick(ChartBrick):
+    """
+    Base brick that draws BarChart.
+    Expects data with the following format :
+    ```
+        [{x: "name", y: value}, ...]
+    ```
+    """
+    template_name = "sketch/bricks/bar-chart.html"
+
+    limits = []
+    enable_transition = True
+
+    def get_chart_props(self, context):
+        return {
+            "limits": self.limits,
+            "transition": self.enable_transition,
+        }
+
+
+class GroupBarChartBrick(ChartBrick):
+    """
+    Base brick that draws GroupBarChart.
+    Expects data with the following format :
+    ```
+        [{x: "name", y: value, group: name}, ...]
+    ```
+    """
+    template_name = "sketch/bricks/groupbar-chart.html"
+    limits = []
+    enable_transition = True
+    enable_legend = True
+
+    def get_chart_props(self, context):
+        return {
+            "limits": self.limits,
+            "transition": self.enable_transition,
+            "showLegend": self.enable_legend,
+        }
+
+
+class StackBarChartBrick(ChartBrick):
+    """
+    Base brick that draws StackBarChart.
+    Expects data with the following format :
+    ```
+        [{x: "name", y: value, group: name}, ...]
```

Comment:
group: name => group: "Name of the group"  ??
What is a group ?

Reply:
do you prefer "z" ?